### PR TITLE
[#39] Complaint features

### PR DESCRIPTION
**Pull request by VladimirBudjen**

### CODE REVIEW

File: src/main/java/rs/ac/uns/ftn/isa/pharmacy/demo/controller/ComplaintController.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @GetMapping("/")
+    @PreAuthorize("hasRole('ROLE_SYSTEM_ADMINISTRATOR')")
+    public ResponseEntity<List<ComplaintDto>> getUnresolvedComplains() {
+        try {
+            return new ResponseEntity<>(complaintService.getUnresolvedComplains(), HttpStatus.OK);
```
Comment (milicasiriski) [inline src/main/java/rs/ac/uns/ftn/isa/pharmacy/demo/controller/ComplaintController.java:58]: unresolvedComplaints

File: src/main/java/rs/ac/uns/ftn/isa/pharmacy/demo/mail/ComplaintMailFormatter.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+                "<hr><br><br> Thank you for using our application!" +
+                "        </p>\n" +
+                "    </body>\n" +
+                "</html>";
+    }
+
+    @Override
+    public String getSubject() {
+        return "Complain answer.";
```
Comment (milicasiriski) [inline src/main/java/rs/ac/uns/ftn/isa/pharmacy/demo/mail/ComplaintMailFormatter.java:28]: Complaint

**Review by lukaDoric — APPROVED** [review]

File: src/main/java/rs/ac/uns/ftn/isa/pharmacy/demo/model/enums/ComplainedSubject.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package rs.ac.uns.ftn.isa.pharmacy.demo.model.enums;
+
+public enum ComplainedSubject {
```
Comment (milicasiriski) [inline src/main/java/rs/ac/uns/ftn/isa/pharmacy/demo/model/enums/ComplainedSubject.java:3]: Complaint subject

File: src/main/java/rs/ac/uns/ftn/isa/pharmacy/demo/repository/PharmacyRepository.java [hunk 4]
```diff
@@ -20,5 +20,24 @@
...
+            "FROM pharmacy AS p, exam AS e, dermatologist_employment_mapping AS dem " +
+            "where p.id=dem.pharmacy_id AND e.patient_id=:patientId"
+            , nativeQuery = true)
+    List<Pharmacy> findPharmacyByPatientIdDermatologistsExam(@Param("patientId") Long patientId);
+
+    @Query(value = "SELECT DISTINCT p.id, p.about, p.city, p.country, p.latitude, p.longitude, p.street," +
+            " p.name, p.pharmacist_exam_duration, p.pharmacist_exam_price, p.rating " +
+            "FROM pharmacy AS p, exam AS e, pharmacy_user AS u " +
+            "where u.id=e.pharmacist_id AND p.id=u.pharmacy_id AND e.patient_id=:patientId", nativeQuery = true)
+    List<Pharmacy> findPharmacyByPatientIdPhysiciansExam(@Param("patientId") Long patientId);
```
Comment (milicasiriski) [inline src/main/java/rs/ac/uns/ftn/isa/pharmacy/demo/repository/PharmacyRepository.java:40]: Pharmacists exam 😄

**Review by milicasiriski — APPROVED** [review]
Good PR! Some names have typos and minor mistakes

**Pull request merged**